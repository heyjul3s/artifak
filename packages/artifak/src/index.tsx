export {
  createStyledComponent,
  createBaseComponents
} from '@artifak/component-generator';

export { BlockBase, createBlockComponents } from '@artifak/block';
export { BlockBaseProps } from '@artifak/block';

export { FlexRow, FlexRowBase, FlexCol, FlexColBase } from '@artifak/flex';
export { FlexRowBaseProps, FlexColBaseProps } from '@artifak/flex';

export { Grid, GridBase, GridItem, GridItemBase } from '@artifak/grid';
export { GridBaseProps, GridItemBaseProps } from '@artifak/grid';

export {
  createTypographyComponents,
  fluidSizing,
  fontWeight,
  TypographyBase
} from '@artifak/typography';
export { TypographyBaseProps } from '@artifak/typography';

export { Imagery, ImageryBase, ImageryProps } from '@artifak/imagery';

export { media, Media } from '@artifak/media';
