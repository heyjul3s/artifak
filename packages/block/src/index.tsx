export { createBlockComponents } from './createBlockComponents';
export { BlockBase } from './BlockBase';
export { BlockBaseProps } from './typings';
