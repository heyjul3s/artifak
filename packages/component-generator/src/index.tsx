export { createComponents } from './createComponents';
export { createStyledComponent } from './createStyledComponent';
export {
  Attrs,
  BaseProps,
  BaseComponentProps,
  GenericRecord,
  Settings,
  StyledComponentConfig,
  Variant
} from './typings';
