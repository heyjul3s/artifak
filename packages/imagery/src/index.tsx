export { Imagery } from './Imagery';
export { ImageryBase } from './ImageryBase';
export { ImageryProps } from './typings';
