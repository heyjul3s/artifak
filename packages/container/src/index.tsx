import React from 'react';
import { createStyledComponent, BaseProps } from '@artifak/component-generator';

export function Container({ children, ...props }) {
  return <ContainerBase {...props}>{children}</ContainerBase>;
}

const ContainerBase = createStyledComponent<BaseProps<any>>({
  styles: {
    margin: '0 auto',
    width: '100%'
  }
});
