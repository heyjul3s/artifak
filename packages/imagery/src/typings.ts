import { BaseComponentProps } from '@artifak/component-generator';

export type ImageryProps = {
  ariaLabel?: string;
  ariaLabelledBy?: string;
  ariaDescribedBy?: string;
  alt?: string;
  crossOrigin?: 'anonymous' | 'use-credentials';
  decoding?: 'async' | 'sync' | 'auto';
  onLoad?: () => void;
  onError?: () => void;
  sizes?: string;
  src: string;
  srcset?: string;
  style?: React.CSSProperties;
} & BaseComponentProps;
