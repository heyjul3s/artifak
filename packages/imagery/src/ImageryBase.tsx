import { StyledComponent } from 'styled-components';
import { system } from 'styled-system';
import { createStyledComponent } from '@artifak/component-generator';
import { ImgComponent } from './typings';

const imageStylesConfig = system({
  objectFit: {
    property: 'objectFit'
  },
  objectPosition: {
    property: 'objectPosition'
  }
});

export const ImageryBase: StyledComponent<
  'img',
  any,
  ImgComponent.Props,
  any
> = createStyledComponent<Partial<ImgComponent.Props>>(
  [imageStylesConfig],
  { display: 'block', width: '100%', height: 'auto' },
  'img'
);
