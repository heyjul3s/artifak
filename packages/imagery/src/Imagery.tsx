import React from 'react';
import { ImageryBase } from './ImageryBase';
import { imgSizes } from './utils/attributes';

/* eslint-disable @typescript-eslint/no-explicit-any */
export const Imagery = React.forwardRef((props: any, ref) => {
  const sizes = imgSizes(props.srcset, props.sizes);

  return <ImageryBase sizes={sizes} {...props} ref={ref} />;
});
