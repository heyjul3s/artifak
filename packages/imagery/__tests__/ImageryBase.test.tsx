import React from 'react';
import { render } from '@testing-library/react';
import { ImageryBase } from '../src/ImageryBase';
import 'jest-styled-components';

describe('ImageryBase', () => {
  it('should render', () => {
    const { asFragment } = render(<ImageryBase />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render styles object-fit and object-position', () => {
    const { asFragment, container } = render(
      <ImageryBase objectFit="cover" objectPosition="center" />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('object-fit', 'cover');
    expect(container.firstChild).toHaveStyleRule('object-position', 'center');
  });
});
