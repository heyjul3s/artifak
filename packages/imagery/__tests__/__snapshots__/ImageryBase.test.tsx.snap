// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageryBase should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa kDzGxq"
  />
</DocumentFragment>
`;

exports[`ImageryBase should render styles object-fit and object-position 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa kDzGxq"
  />
</DocumentFragment>
`;
